requirements.txt

```diff
@@ -4,8 +4,7 @@ v3io~=0.5.14
 # and 1.5.* due to https://github.com/pandas-dev/pandas/issues/49203
 pandas>=1, !=1.5.*, <3
 numpy>=1.16.5,<1.23
-# pyarrow 13 and over cause test failures
-pyarrow>=1,<13
+pyarrow>=1,<15
```

Comment:
Maybe just mention that this is a safeguard. I.e. that this limit isn't because there's something wrong with pyarrow 15 (which doesn't even exist yet).

Reply:
Actually I would to revisit our practice of setting an upper limit on some later major version.
With the current approach, frequently we will get a request to upgrade from the field (like now) before we understand there is an issue in supporting the later version. In some cases - the user will need to wait for the next mlrun version.
On the other hand if we remove the upper limit, chances are we will hit the bug before the user, and the user will have an mlrun that support the new version by the time it needs it.
what do you think?

Reply:
Only not sure it's true that we'll hit this before the user... And if we do, we won't be able to add the limit after the fact. Without the limit, a new release could immediately introduce an issue for the user, as well as break CI (which wouldn't be so bad, but it's sometimes hard to realize that CI was broken by a new version of a dependency and not by a PR or sporadic failure or env issue).

Does the user actually need to wait for a new release? They should be able to install whatever library they want and only get a warning regarding the incompatibility.

Reply:
I see, I guess will make more sense once we have dependencies freeze.

Reply:
@gtopper , what about padnas<3? Also a comment?  Should be more generic approach

Reply:
In principle you're right @alxtkr77. In practice, pyarrow releases are much more frequent, which is why the limit is there. A general guideline would be great, but maybe the dependency freeze takes care of it anyway.

Reply:
Will add as requested